### PR TITLE
Dns integration

### PR DESCRIPTION
**Pull request by lazypower**
this is a hack, take a look plz

### CODE REVIEW

File: files/dns/nginx-vhost [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# This file is managed by Juju and follows the Template Language from Consul:
+# https://github.com/hashicorp/consul-template#templating-language
+
+{{range services}}
+upstream {{.Name}} {
```
Comment (lazypower) [inline files/dns/nginx-vhost:5]: Defines the backend service in docker

File: files/dns/nginx-vhost [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    {{end}}
+}
+
+
+server {
+  listen 80 {{.Name}}.domain;
+
+  charset utf-8;
+
+  location / {
```
Comment (lazypower) [inline files/dns/nginx-vhost:17]: This is the "server" config, which denotes which DNS entry to listen for, and which upstream block to consume.

File: README.md [hunk 3]
```diff
@@ -18,15 +17,25 @@ Step by step instructions on deploying this charm:
...
-# Contact Information
+### Reconfigure w/ auto forwarding from Consul
 
-Though this will be listed in the charm store itself don't assume a user will know that, so include that information here:
+The charm has a `public-website` relationship, that implements a prototype
+interface to setup automatic website proxy'ing based on data from Consul.
 
+    juju deploy cs:~zoology/trusty/consul
+    juju deploy cs:~zoology/trusty/dns
+    juju set dns provider=rt53 provider_keys="AWS_ACESS_KEY_ID|XXX AWS_PRIVATE ACCESS_KEY|XXX"
```
Comment (mbruzek) [inline README.md:32]: AWS_ACCESS_KEY is misspelled here.  Also this is kind of a complex data structure to set using Juju.  You will need more explanation when to use spaces or pipes.

File: hooks/public-website-relation-changed [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    domain = hookenv.relation_get('domain')
+    if domain:
+        # we have dns data, update existing consul-template
+        # TODO: make this more permanent and live w/ the template relation
+        with open('files/dns/nginx-vhost', 'r') as f:
+            template = f.read()
+            template = template.replace('.domain', ".{}".format( domain))
+        with open('/etc/nginx/nginx.ctmpl', 'w+') as f:
+            f.write(template)
+        subprocess.call(['service','consul-template','restart'])
```
Comment (mbruzek) [inline hooks/public-website-relation-changed:33]: There is a charm-helper for this.  Why not use it?  host.service (restart or reload)

**Comment by mbruzek** [comment]
I noted a few minor points in line.  The relation name gives me a bit of worry because we already have the http relation, so I would be confused about the public-http relation being a new user.

You have my +1 to merge this code for now.  We should revisit this code as our use cases and needs become more clear.

**Pull request merged**